Core/src/Surfaces/Surface.cpp

```diff
@@ -22,21 +22,26 @@ std::array<std::string, Acts::Surface::SurfaceType::Other>
         "Cone", "Cylinder", "Disc", "Perigee", "Plane", "Straw", "Curvilinear"};
 
 Acts::Surface::Surface(const Transform3& transform)
-    : GeometryObject(), m_transform(transform) {}
+    : GeometryObject(), m_transform(std::make_unique<Transform3>(transform)) {}
 
 Acts::Surface::Surface(const DetectorElementBase& detelement)
     : GeometryObject(), m_associatedDetElement(&detelement) {}
 
 Acts::Surface::Surface(const Surface& other)
     : GeometryObject(other),
       std::enable_shared_from_this<Surface>(),
-      m_transform(other.m_transform),
-      m_surfaceMaterial(other.m_surfaceMaterial) {}
+
+      m_associatedDetElement(other.m_associatedDetElement),
+      m_surfaceMaterial(other.m_surfaceMaterial) {
+  if (other.m_transform) {
+    m_transform = std::make_unique<Transform3>(*other.m_transform);
+  }
```

Comment:
this sort of handling makes me wonder if a `std::shared_ptr` would be the better option

Reply:
I don't think we copy surfaces very often.

Reply:
Right I also don't think it would result in any significant memory reduction but it might make the code a bit easier to read

Reply:
so shared or unique? That's the question ;)

Reply:
Let's leave it as it is. I don't think this is a critical thing and it is just a personal preference on my side